src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs

```diff
@@ -382,6 +382,51 @@ void UpdateLeftToolbarItems()
 			UpdateBackButtonTitle();
 		}
 
+		string GenerateImageHash(UIImage image)
+		{
+			if (image == null)
+				return string.Empty;
+			using (NSData imageData = image.AsPNG())
+			{
+				using (var sha256 = System.Security.Cryptography.SHA256.Create())
+				{
+					byte[] hashBytes = sha256.ComputeHash(imageData.ToArray());
+					return Convert.ToBase64String(hashBytes);
+				}
+			}
+		}
+		UIImage ResizeImage(UIImage sourceImage, CGSize targetSize)
```

Comment:
This creates a new image with new data, and potentially wastes memory. I feel like we have this problem solved somewhere before that just scales the image. Let me see...

Reply:
This place: https://github.com/dotnet/maui/blob/af94a2246539089d8ec82944d4b0c2afee383de4/src/Controls/src/Core/Button/Button.iOS.cs#L433-L449

I think we can maybe extract this into some helper class, maybe an extension for UIImage?

Reply:
Also, we probably want to preserve aspect ratio, the code `sourceImage.Draw(new CGRect(0, 0, targetSize.Width, targetSize.Height));` will stretch in this PR. So the scaling way we have in button is probably safer.

Reply:
@mattleibow, Thank you for the suggestion. I have moved the ResizeImageSource method from Button.iOS to the UIImageExtensions class. I applied the same logic in the ShellPageRendererTracker for the custom FlyoutIcon and also updated the button related code. Could you please review these changes and let me know if you have any concerns?